src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,78 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
+import cn from 'classnames';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { actions } from '../../features/currentTodo';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+export const TodoModal: FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
 
-      <Loader />
+  useEffect(() => {
+    if (!currentTodo || currentTodo.userId === user?.id) {
+      return;
+    }
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+    getUser(currentTodo.userId)
+      .then(item => setUser(item))
+      .catch(() => {
+        throw new Error('Something went wrong');
```

Comment:
Consider setting the error into the state or redux state instead of throwing the error